daemon/api.go

```diff
@@ -1225,3 +1233,41 @@ func abortChange(c *Command, r *http.Request, user *auth.UserState) Response {
 
 	return SyncResponse(change2changeInfo(chg), nil)
 }
+
+var (
+	ssoCreateUser                = sso.CreateUser
+	postCreateUserUcrednetGetUID = ucrednetGetUID
+)
+
+func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response {
+	uid, err := postCreateUserUcrednetGetUID(r.RemoteAddr)
+	if err != nil {
+		return BadRequest("cannot get ucrednet uid: %v", err)
+	}
+	if uid != 0 {
+		return BadRequest("cannot use create-user as non-root")
+	}
+
+	var createData struct {
+		EMail string `json:"email"`
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&createData); err != nil {
+		return BadRequest("cannot decode create-user data from request body: %v", err)
+	}
+
+	// FIXME: this will block, we really need to make this an
```

Comment:
The request is running inside a goroutine on the server end, and there's an expectation that the task will run quite quickly, or it should fail. It seems fine for it to only respond after the task being accomplished.